memory/model/nearest_neighbors/neighbor_gen.py

```diff
@@ -0,0 +1,39 @@
+from nearpy import Engine
+from nearpy.hashes import RandomBinaryProjections
+from nearpy.filters import NearestFilter
+import numpy as np
+import os
+
+class NeighborGenerator:
+    def __init__(self, dimension=9984, num_neighbors=10, distance_fxn=None):
+        self.dimension = dimension
+        self.num_neighbors = num_neighbors
+        self.distance_fxn = distance_fxn
+        if not distance_fxn:
+            self.engine = Engine(9984)
+
+    def load_data(self, dir):
+        # expects dir to be in same format that siamese network uses, but no view folder
+        # takes in npz files produced by featurization script
+        for obj in os.listdir(dir):
+            obj_dir = os.path.join(os.path.join(dir, obj), 'view_000000')
+            for file in os.listdir(obj_dir):
+                image = os.path.join(obj_dir, file)
+                image = np.load(image)['arr_0']
+                self.engine.store_vector(image, obj)
+
+    def predict(self, file_location):
```

Comment:
Can we have this and the following `batch_predict` take in arrays of inputs to query instead of directories? In general, try to keep operations on directories at the highest level of your code (such as in the `tools/` scripts), and use array abstractions for lower levels.